src/sql/azdata.proposed.d.ts

```diff
@@ -1890,6 +1890,48 @@ declare module 'azdata' {
 		 */
 		export function openCustomErrorDialog(options: IErrorDialogOptions): Thenable<string | undefined>;
 
+		/**
+		 * Opens the troubleshooter dialog with customization options provided.
+		 * @param options Dialog options to customize troubleshooter dialog.
+		 * @returns Id of action button clicked by user, e.g. ok, cancel
+		 */
+		export function openTroubleshooterDialog(options: ITroubleshooterDialogOptions): Thenable<string | undefined>;
+
+
+		/**
+		 * State of the troubleshooter item
+		 */
+		export enum State {
+			Invalid = 0,
+			Loading = 1,
+			Valid = 2
+		}
+
+		/**
+		 * Troubleshooter item to display in the troubleshooter dialog
+		 */
+		export interface ITroubleshooterItem {
+			/**
+			 * State of fetching the troubleshooter item
+			 */
+			state: State;
```

Comment:
What is this state supposed to be doing? It doesn't appear to be used at all currently, and it's not clear to me from the description what it'll be controlling later on.

If you aren't going to add the implementation for that right now it'd probably be best to just remove it for now and add it back in later once you have that hooked up.

Reply:
So I see you've hooked up state stuff, but how are extensions supposed to use this? It seems like they can only call openTroubleshooterDialog directly, and after that won't be able to update the state. Is there some other functionality you're planning on adding here to allow them to change the state of an open dialog?

Reply:
As of now we haven't designed that, so it's a work in progress, that's why going in feature branch - when completed, we would be able to update state after receiving full results, so the state icon and results will be updated after results arrive.